Source/WebCore/style/StyleTreeResolver.cpp

```diff
@@ -1656,14 +1656,17 @@ void TreeResolver::sortPositionOptionsIfNeeded(PositionOptions& options, const S
 
     // If the styleable has a last successful position option...
     if (auto lastSuccessfulIndex = m_document->styleScope().lastSuccessfulPositionOptionIndexFor(styleable)) {
-        // ... we look for the last successful fallback ...
-        for (size_t i = 1; i < options.optionStyles.size(); ++i) {
-            // ... if this option is the last successful fallback (as indicated by the index) ...
-            if (options.optionStyles[i].style->usedPositionOptionIndex() == *lastSuccessfulIndex) {
-                // ... rotate the options array to make it the original style (at index 0).
-                std::ranges::rotate(options.optionStyles, &options.optionStyles[i]);
-                break;
-            }
+        // ... find which style in options.optionStyles has that index
+        auto lastSuccessfulIndexInOptionStyles = options.optionStyles.findIf([lastSuccessfulIndex] (const auto& option) {
+            ASSERT(option.style);
+            return option.style->usedPositionOptionIndex() == *lastSuccessfulIndex;
+        });
+
+        // If there's one, move it to the beginning.
+        if (lastSuccessfulIndexInOptionStyles != WTF::notFound) {
+            auto lastSuccessfulOption = WTFMove(options.optionStyles[lastSuccessfulIndexInOptionStyles]);
+            options.optionStyles.removeAt(lastSuccessfulIndexInOptionStyles);
+            options.optionStyles.insert(0, WTFMove(lastSuccessfulOption));
         }
```

Comment:
Question: Would it make sense to do it like this, or does it not make sense?

```suggestion
        if (lastSuccessfulIndexInOptionStyles > 0) {
            options.optionStyles.insert(0, WTFMove(options.optionStyles[lastSuccessfulIndexInOptionStyles]);
            options.optionStyles.removeAt(lastSuccessfulIndexInOptionStyles);
        }
```

This change would do two things, assuming it works as I think:
- Skips the move if it's already at the front.
- Only does one WTFMove operation.

Reply:
If you do like the suggestive change then it should be
```
options.optionStyles.removeAt(lastSuccessfulIndexInOptionStyles + 1);
```

Because you add a new element in the front, so now the index of the option is + 1. But then it's a bit hard to read. I'll take the `lastSuccessfulIndexInOptionStyles > 0` though.